pvlib/tracking.py

```diff
@@ -57,7 +57,13 @@ class SingleAxisTracker(PVSystem):
         `cross_axis_tilt`. [degrees]
 
     **kwargs
-        Passed to :py:class:`~pvlib.pvsystem.PVSystem`.
+        Passed to :py:class:`~pvlib.pvsystem.PVSystem`. If the `arrays`
+        parameter is specified it must have only a single Array.
```

Comment:
Reasonable given the current object hierarchy, but also makes me think the hierarchy needs to change.

Reply:
If my refactor is rejected as out of scope, we at least need to prevent this kind of non-nonsensical thing:

```python
>>> tracking.SingleAxisTracker(arrays=[pvsystem.Array(surface_tilt=30, surface_azimuth=180)])
SingleAxisTracker:
  axis_tilt: 0
  axis_azimuth: 0
  max_angle: 90
  backtrack: True
  gcr: 0.2857142857142857
  cross_axis_tilt: 0.0
  name: None
  Array:
    name: None
    surface_tilt: 30
    surface_azimuth: 180
    module: None
    albedo: 0.25
    racking_model: None
    module_type: None
    temperature_model_parameters: {}
    strings: 1
    modules_per_string: 1
  inverter: None
```

Perhaps require that an `Array` passed to `SingleAxisTracker` has `surface_tilt=None` and `surface_azimuth=None`.

Reply:
I've addressed this by raising a `ValueError` for the case demonstrated above. The original behavior (pre `Array` class) was to override the `surface_tilt` and `surface_azimuth` parameters; however, for an `Array` input I didn't want to modify an object that a user could have other references to. I think that is the right thing to do, but just wanted to be clear about it and my reasons.